zerver/lib/import_realm.py

```diff
@@ -444,6 +452,53 @@ def fix_message_rendered_content(
             )
 
 
+def valid_near_link_parts(near_link_parts: list[str]) -> bool:
+    return (
+        3 <= len(near_link_parts) <= 7
+        and near_link_parts[0] == "#narrow"
+        and near_link_parts[1] in ["channel", "stream"]
+    )
```

Comment:
The techniques used in `is_same_server_message_link` for URL parsing would be considerably more suitable for this sort of logic.

Reply:
I ended up refactoring some logic out of `is_same_server_message_link` and consolidated it with this check, into a new function, `check_near_link_base`. This refactor is in aca6115c073375273391b176c70eadd6e7e4dc80

Reply:
This seems solid.